pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -104,19 +105,30 @@ int parse_iso_8601_datetime(const char *str, int len, int want_exc,
     while (sublen > 0 && isspace(*substr)) {
         ++substr;
         --sublen;
+        if (format_len < 1 && exact) goto parse_error;
```

Comment:
```suggestion
        if (format_len < 1) {
          if (exact) {
            goto parse_error;
          }
        // Do something?
        } else {
          if (*format++ != ' ') {
            goto parse_error;
          }
          
          format_len--;
        }
```

I think this control flow would work more universally and avoid logic pitfalls / segfaults (see other comment)